tests/system/hw_math/source/main.c

```diff
@@ -230,14 +417,271 @@ int main(int argc, char **argv)
+    }
+
+
+    {
+           uint32_t sw_iterations = 0;
+           uint32_t sw_time = 0;
+
+           cpuStartTiming(0); // This uses timers 0 and 1
+
+           for (uint32_t i = 0; i< TEST_SIZE; ++i)
+           {
+
+                sw_iterations++;
+                // Write to a volatile register to prevent the compiler from
+                // optimizing out the operation (it doesn't have any side-effect, so
+                // the compiler would probably remove it without the write).
+                vu = isqrt64( (((uint64_t )i)<<32)| i );
+
+           }
+
+            sw_time += cpuEndTiming();
+
+            printf("rosetta soft sqrt64 time: %lu (%lu per op)\n", sw_time, sw_time / sw_iterations);
+    }
+
+
+
+    {
+            uint32_t sw_iterations = 0;
+            uint32_t sw_time = 0;
+
+            cpuStartTiming(0); // This uses timers 0 and 1
+
+            for (uint32_t i = 0; i< TEST_SIZE; ++i)
+            {
+
+                sw_iterations++;
+                // Write to a volatile register to prevent the compiler from
+                // optimizing out the operation (it doesn't have any side-effect, so
+                // the compiler would probably remove it without the write).
+                slow_sqrt64( (((uint64_t )i)<<32)| i );
+                vu = sqrt64_result();
+
+
+             }
+
+            sw_time += cpuEndTiming();
+
+            printf("slow_sqrt64 time: %lu (%lu per op)\n", sw_time, sw_time / sw_iterations);
+    }
+    {
+            uint32_t sw_iterations = 0;
+            uint32_t sw_time = 0;
+
+            cpuStartTiming(0); // This uses timers 0 and 1
+
+            for (uint32_t i = 0; i< TEST_SIZE; ++i)
+            {
+
+                sw_iterations++;
+                // Write to a volatile register to prevent the compiler from
+                // optimizing out the operation (it doesn't have any side-effect, so
+                // the compiler would probably remove it without the write).
+                fast_sqrt64( (((uint64_t )i)<<32)| i );
+                vu = sqrt_result();
+
+
+            }
+
+            sw_time += cpuEndTiming();
+
+            printf("Fast sqrt64 time: %lu (%lu per op)\n", sw_time, sw_time / sw_iterations);
+    }
+
+
+
+
+    {
+            uint32_t sw_iterations = 0;
+            uint32_t sw_time = 0;
+
+            cpuStartTiming(0); // This uses timers 0 and 1
+
+            for (uint32_t i = 0; i< TEST_SIZE; ++i)
+            {
+
+                sw_iterations++;
+                // Write to a volatile register to prevent the compiler from
+                // optimizing out the operation (it doesn't have any side-effect, so
+                // the compiler would probably remove it without the write).
+                sqrt64_asynch( (((uint64_t )i)<<32)| i );
+                vu = sqrt64_result();
+
+
+            }
+
+            sw_time += cpuEndTiming();
+
+            printf("BlocksDS sqrt64 time: %lu (%lu per op)\n", sw_time, sw_time / sw_iterations);
```

Comment:
This won't fit in one line, you should split it into two so that numbers aren't cut in the middle.